imap_processing/mag/l1c/mag_l1c.py

```diff
@@ -0,0 +1,52 @@
+"""MAG L1C processing module."""
+
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+
+
+def mag_l1c(
+    first_input_dataset: xr.Dataset, second_input_dataset: xr.Dataset, version
+) -> xr.Dataset:
+    """
+    Process MAG L1C data from L1A data.
+
+    This requires both the norm and burst data to be passed in.
+
+    Parameters
+    ----------
+    first_input_dataset : xr.Dataset
+        The first input dataset to process. This can be either burst or norm data, for
+        mago or magi.
+    second_input_dataset : xr.Dataset
+        The second input dataset to process. This should be burst if first_input_dataset
+        was norm, or norm if first_input_dataset was burst. It should match the
+        instrument - both inputs should be mago or magi.
+    version : str
+        The version of the output data
+    """
+    # TODO: L1C processing involves filling gaps with burst data.
+    input_logical_source_1 = first_input_dataset.attrs["Logical_source"]
```

Comment:
I probably didn't look at this code long enough to see why that won't work, but could you pass in the two datasets as one list and get the index of of the one with norm below and use that to index the list? Something like

```python
norm_idx = [index for index, value in enumerate(data_list) if 'norm' in value]
data_list[norm_idx[0]]
```

Reply:
Maybe? Probably? I think because the logical source is sometimes a list and sometimes not, this approach would end up being difficult to parse.